src/test/java/uk/gov/hmcts/reform/sscs/service/ServiceHearingsServiceTest.java

```diff
@@ -0,0 +1,191 @@
+class ServiceHearingsServiceTest {
+
+    private static final long CASE_ID = 12345L;
+
+    @Mock
+    private SessionCategoryMapService sessionCategoryMaps;
+
+    @Mock
+    private HearingDurationsService hearingDurations;
+
+    @Mock
+    private VenueService venueService;
+
+    @Mock
+    private ReferenceDataServiceHolder referenceDataServiceHolder;
+
+    @Mock
+    private CcdCaseService ccdCaseService;
+
+    @InjectMocks
+    private ServiceHearingsService serviceHearingsService;
+
+    private SscsCaseData caseData;
+    private SscsCaseDetails caseDetails;
+
+
+    @BeforeEach
+    void setup() {
+        caseData = SscsCaseData.builder()
+            .ccdCaseId("1234")
+            .benefitCode(BENEFIT_CODE)
+            .issueCode(ISSUE_CODE)
+            .urgentCase("Yes")
+            .adjournCaseCanCaseBeListedRightAway("Yes")
+            .caseManagementLocation(CaseManagementLocation.builder()
+                .baseLocation("LIVERPOOL SOCIAL SECURITY AND CHILD SUPPORT TRIBUNAL")
+                .region("North West")
+                .build())
+            .appeal(Appeal.builder()
+                .hearingType("final")
+                .appellant(Appellant.builder()
+                    .name(Name.builder()
+                        .firstName("Fred")
+                        .lastName("Flintstone")
+                        .title("Mr")
+                        .build())
+                    .build())
+                .hearingSubtype(HearingSubtype.builder()
+                    .hearingTelephoneNumber("0999733733")
+                    .hearingVideoEmail("test@gmail.com")
+                    .wantsHearingTypeFaceToFace("Yes")
+                    .wantsHearingTypeTelephone("No")
+                    .wantsHearingTypeVideo("No")
+                    .build())
+                .hearingOptions(HearingOptions.builder()
+                    .wantsToAttend("Yes")
+                    .build())
+                .rep(Representative.builder()
+                    .hasRepresentative("Yes")
+                    .name(Name.builder()
+                        .title("Mr")
+                        .firstName("Harry")
+                        .lastName("Potter")
+                        .build())
+                    .build())
+                .build())
+            .languagePreferenceWelsh("No")
+            .linkedCasesBoolean("No")
+            .sscsIndustrialInjuriesData(SscsIndustrialInjuriesData.builder()
+                .panelDoctorSpecialism("cardiologist")
+                .secondPanelDoctorSpecialism("eyeSurgeon")
+                .build())
+            .build();
+
+        caseDetails = SscsCaseDetails.builder()
+            .data(caseData)
+            .build();
+    }
+
+    @DisplayName("When a case data is retrieved an entity which does not have a Id, that a new Id will be generated and the method updateCaseData will be called once")
+    @Test
+    void testGetServiceHearingValuesNoIds() throws Exception {
```

Comment:
minor/optional: since we're setting up the same mock data for both tests - can we put them in the setup method?